### ISSUE TITLE
PAL3 API Update

### ISSUE THREAD

**Issue by jonwetzel** [Dragoon-Lab/LaitsV3#79]
The following changes from Benjamin Nye should be done on dragoon.asu.edu/PAL3.

API Update for Working With External Active Resources (e.g., Dragoon): Based on some new needs for PAL3, we had to do some small changes to the API.  The most substantial change was that we now have multiple backend servers for different purposes (e.g., a main one, a dev, one for just small studies/prototying).  Currently, Dragoon doesn't know anything about that and just sends its scores to the main PAL3 server, regardless of which one invokes Dragoon.

We have changed the messaging slightly to pass this information so that Dragoon could respond back properly.  Do you think that you would be able to do the minor changes that would be needed to have this message properly?  We still occasionally demo Dragoon in PAL3 to Navy folks and others and highlight your system and what it does, but it gets tricky because we can only do it on PAL3 builds that point to a certain server.

I'm pasting the notes about the new API.  You've already got the basic messaging in place, so this would just be some small edits to the existing REST call (nothing huge, and Adam can support on this):

For URL activities, we send data via URL parameters.  For the data to PAL3, we receive it by a REST endpoint. Right now we are still relying on a lightweight version that is easier to deal with, though we're likely to replace this with official xAPI stuff down the road.

SENT VIA URL PARAMS: When a resource is started from pal, it is sent the following data:
- player_id
- player_name
- resource_guid
- resource_session_id (the specific instance of this resource for this player)*
- database_url (where the data will be stored)*

RECEIVED VIA REST: Pal Receives, When a resource is completed, pal receives the following data
- score
- player_id
- resource_guid
- resource_session_id*
- duration (how long it took to complete)*
Optional:
- Pair<knowledge component id, score> (these scores override the previously listed score)*
(you can ignore the KC: score optional one, unless you are giving different scores for different KC's)

So in short, we send everything at we need back, except for the duration and the score.  You can send us the specific KC's that you're using (though we'd need to make sure you know our ID's for them), but the easiest approach right now is that we register the resources with our ID's.
